src/Persistence_matrix/concept/PersistenceMatrixColumn.h

```diff
@@ -0,0 +1,470 @@
+   * if @ref PersistenceMatrixOptions::is_z2 is false, also when using
+   * @ref Cell_field_element::get_element "Cell::get_element".
+   * @return Reference to this column.
+   */
+  template <class Cell_range>
+  PersistenceMatrixColumn& multiply_and_add(const Field_element_type& val, const Cell_range& column);
+  /**
+   * @brief `this = val * this + column`
+   * 
+   * @param val Value to multiply.
+   * @param column Column to add.
+   * @return Reference to this column.
+   */
+  PersistenceMatrixColumn& multiply_and_add(const Field_element_type& val, PersistenceMatrixColumn& column);
+  /**
+   * @brief `this = this + column * val`
+   * 
+   * @tparam Cell_range @ref Cell range with %begin() and %end() method.
+   * @param column @ref Cell range. Every cell has to return the right value when using @ref Cell::get_row_index and,
+   * if @ref PersistenceMatrixOptions::is_z2 is false, also when using
+   * @ref Cell_field_element::get_element "Cell::get_element".
+   * @param val Value to multiply.
+   * @return Reference to this column.
+   */
+  template <class Cell_range>
+  PersistenceMatrixColumn& multiply_and_add(const Cell_range& column, const Field_element_type& val);
+  /**
+   * @brief `this = this + column * val`
+   * 
+   * @param column Column to add.
+   * @param val Value to multiply.
+   * @return Reference to this column.
+   */
+  PersistenceMatrixColumn& multiply_and_add(PersistenceMatrixColumn& column, const Field_element_type& val);
+
+  /**
+   * @brief Equality comparator. Equal in the sense that what is "supposed" to be contained in the columns is equal,
+   * not what is actually stored in the underlying container. For exemple, the underlying container of 
+   * @ref Vector_column can contain cells which were erased explicitely by @ref clear(index). Those cells should not
+   * be taken into account while comparing.
+   * 
+   * @param c1 First column to compare.
+   * @param c2 Second column to compare.
+   * @return true If both column are equal.
+   * @return false Otherwise.
+   */
+  friend bool operator==(const PersistenceMatrixColumn& c1, const PersistenceMatrixColumn& c2);
+  /**
+   * @brief "Strictly smaller than" comparator. Usually a lexicographical order, but what matters is that the
+   * order is total. The order should apply on what is "supposed" to be contained in the columns,
+   * not what is actually stored in the underlying container. For exemple, the underlying container of 
+   * @ref Vector_column can contain cells which were erased explicitely by @ref clear(index). Those cells should not
+   * be taken into account while comparing.
+   * 
+   * @param c1 First column to compare.
+   * @param c2 Second column to compare.
+   * @return true If the first column is strictly smaller than the second one.
+   * @return false Otherwise.
+   */
+  friend bool operator<(const PersistenceMatrixColumn& c1, const PersistenceMatrixColumn& c2);
```

Comment:
I am not sure to understand the question. If I remember well, I mainly use this for column compression, so as you mention for a `find` and duplicate detection inside a boost intrusive set. But it doesn't work as well as we want, as the comparator is usually linear and is needed too often. If you have any ideas on how to fasten up the column compression, it would be great!

Reply:
I wanted to be sure I understood that the only purpose of this `operator<` was as a comparator in `std::set` or equivalent, and that you never relied on its order for anything (except being an arbitrary total order).
```
    // and insert it in the matrix, in constant time thanks to the hint cam_.end().
    // Indeed *new_col has the biggest lexicographic value because key is the
    // biggest key used so far.
    cam_.insert(cam_.end(), *new_col);
```
It looks like the current code kind of expects a specific order.

Reply:
Hm, true, I need to re-look at this. I forgot about it when I did the documentation.